bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/flamingos2/F2ScienceAreaGeometry.scala

```diff
@@ -0,0 +1,109 @@
+package edu.gemini.spModel.gemini.flamingos2
+
+import java.awt.Shape
+import java.awt.geom.{Rectangle2D, Area, Ellipse2D}
+
+import edu.gemini.shared.util.immutable.DefaultImList
+import edu.gemini.spModel.gemini.flamingos2.Flamingos2.{FPUnit, LyotWheel}
+import edu.gemini.spModel.inst.ScienceAreaGeometry
+
+import scala.collection.JavaConverters._
+
+class F2ScienceAreaGeometry(inst0: Flamingos2) extends ScienceAreaGeometry[Flamingos2] {
+  import F2ScienceAreaGeometry._
+
+  override def geometry: List[Shape] = {
+    Option(inst0).toList.flatMap { inst =>
```

Comment:
This is `filter` + `map` so you could rewrite it in terms of `collect` but I think it's fine as-is.